src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -619,12 +621,17 @@ namespace smt::noodler {
         conjuncts.push_back(conv_form_with_precision.first);
         precision = conv_form_with_precision.second;
 
-
         // get the LIA formula describing solutions for special predicates
         conjuncts.push_back(get_formula_for_ca_diseqs());
         auto not_cont_prec = get_formula_for_not_contains();
+
         if (not_cont_prec.second == LenNodePrecision::PRECISE || not_cont_prec.second == precision) {
```

Comment:
I think this whole if-else construct should be replaced with
```
conjuncts.push_back(not_cont_prec.first);
precision = get_resulting_precision_for_conjunction(precision, not_cont_prec.second);
```

Reply:
Can you check it? I tried to make things clearer.

Reply:
I still do not understand the if-else that is left there. If there are disequations, we return underapproximating FALSE, what does it mean?

Reply:
I completely removed the condition as it doesn't make sense to me.